src/Microsoft.AspNetCore.Mvc.Abstractions/ModelBinding/IModelBindingContext.cs

```diff
@@ -0,0 +1,149 @@
+        ModelMetadata ModelMetadata { get; set; }
+
+        /// <summary>
+        /// Gets or sets the name of the model. This property is used as a key for looking up values in
+        /// <see cref="IValueProvider"/> during model binding.
+        /// </summary>
+        string ModelName { get; set; }
+
+        /// <summary>
+        /// Gets or sets the <see cref="ModelStateDictionary"/> used to capture <see cref="ModelState"/> values
+        /// for properties in the object graph of the model when binding.
+        /// </summary>
+        ModelStateDictionary ModelState { get; set; }
+
+        /// <summary>
+        /// Gets the type of the model.
+        /// </summary>
+        /// <remarks>
+        /// The <see cref="ModelMetadata"/> property must be set to access this property.
+        /// </remarks>
+        Type ModelType { get; }
+
+        /// <summary>
+        /// Represents the <see cref="OperationBindingContext"/> associated with this context.
+        /// </summary>
+        OperationBindingContext OperationBindingContext { get; set; }
+
+        /// <summary>
+        /// Gets or sets a predicate which will be evaluated for each property to determine if the property
+        /// is eligible for model binding.
+        /// </summary>
+        Func<IModelBindingContext, string, bool> PropertyFilter { get; set; }
+
+        /// <summary>
+        /// Gets or sets the <see cref="ValidationStateDictionary"/>. Used for tracking validation state to
+        /// customize validation behavior for a model object.
+        /// </summary>
+        ValidationStateDictionary ValidationState { get; set; }
+
+        /// <summary>
+        /// Gets or sets the <see cref="IValueProvider"/> associated with this context.
+        /// </summary>
+        IValueProvider ValueProvider { get; set; }
+
+        /// <summary>
+        /// <para>
+        /// On completion returns a <see cref="ModelBindingResult"/> which
+        /// represents the result of the model binding process.
+        /// </para>
+        /// <para>
+        /// If model binding was successful, the <see cref="ModelBindingResult"/> should be a value created
+        /// with <see cref="ModelBindingResult.Success"/>. If model binding failed, the
+        /// <see cref="ModelBindingResult"/> should be a value created with <see cref="ModelBindingResult.Failed"/>.
+        /// If there was no data, or this model binder cannot handle the operation, the
+        /// <see cref="ModelBindingResult"/> should be null.
+        /// </para>
+        /// </summary>
+        ModelBindingResult? Result { get; set; }
+
+        /// <summary>
+        /// Pushes a layer of state onto this context. Model binders will call this as part of recursion when binding properties
+        /// or collection items.
+        /// </summary>
+        /// <param name="modelMetadata"><see cref="ModelMetadata"/> to assign to the <see cref="IModelBindingContext.ModelMetadata"/> property</param>
```

Comment:
`IModelBindingContext.` shouldn't be needed in any of these comments. But first `ModelMetadata` should be either `Value` (consistent with other `<param>`s) or `ModelBinding.ModelMetadata`.